bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/storage/ldb/SingleDirectoryDbLedgerStorage.java

```diff
@@ -469,7 +474,9 @@ private void fillReadAheadCache(long orginalLedgerId, long firstEntryId, long fi
             int count = 0;
             long size = 0;
 
-            while (count < readAheadCacheBatchSize && currentEntryLogId == firstEntryLogId) {
+            while (count < readAheadCacheBatchSize
+                    && size < readAheadCacheBatchSize
```

Comment:
```suggestion
                    && size < maxReadAheadBytesSize
```

Reply:
Good catch, I messed this up when resolving the conflicts...